pkg/resources/nifi/secretconfig.go

```diff
@@ -165,6 +165,8 @@ func (r *Reconciler) getNifiPropertiesConfigString(nConfig *v1.NodeConfig, id in
 		"IsNode":                  nConfig.GetIsNode(),
 		"ZookeeperConnectString":  r.NifiCluster.Spec.ZKAddress,
 		"ZookeeperPath":           r.NifiCluster.Spec.GetZkPath(),
+		"ClusterManager":          r.NifiCluster.Spec.ClusterManager,
+		"KubernetesLeasePrefix":   r.NifiCluster.Name,
```

Comment:
Is there potential for these to conflict if multiple clusters are deployed with the same name (same name across multiple namespaces)? Would it be possible for the operator delete leases/configmaps for another cluster?

Reply:
It should not, because the clean up only occurs on the resources with the corresponding prefix (cluster name) in the corresponding namespace (cluster namespace). And it should be impossible to have 2 NifiClusters with the same name in the same cluster.

```go
func (r *NifiClusterReconciler) deleteLeases(ctx context.Context, cluster *v1.NifiCluster) error {
	clusterCoordinatorName := fmt.Sprintf("%s-cluster-coordinator", cluster.Name)
	primaryNode := fmt.Sprintf("%s-primary-node", cluster.Name)
	foundLeaseList := &coordinationv1.LeaseList{}
	if err := r.Client.List(ctx, foundLeaseList, client.ListOption(client.InNamespace(cluster.Namespace))); err != nil {
		return err
	}
	for _, lease := range foundLeaseList.Items {
		if lease.Name == clusterCoordinatorName || lease.Name == primaryNode {
			r.Log.Debug("Deleting Lease as it matches the name.", zap.String("clusterName", cluster.Name), zap.String("leaseName", lease.Name))
			if err := r.Client.Delete(ctx, &lease); err != nil {
				return err
			}
		}
	}
	return nil
}

func (r *NifiClusterReconciler) deleteConfigMaps(ctx context.Context, cluster *v1.NifiCluster) error {
	nifiComponentPrefix := fmt.Sprintf("%s-nifi-component", cluster.Name)
	foundConfigMapList := &corev1.ConfigMapList{}
	if err := r.Client.List(ctx, foundConfigMapList, client.ListOption(client.InNamespace(cluster.Namespace))); err != nil {
		return err
	}
	for _, configMap := range foundConfigMapList.Items {
		if strings.HasPrefix(configMap.Name, nifiComponentPrefix) {
			r.Log.Debug("Deleting ConfigMap as it matches the prefix.", zap.String("clusterName", cluster.Name), zap.String("configMapName", configMap.Name))
			if err := r.Client.Delete(ctx, &configMap); err != nil {
				return err
			}
		}
	}
	return nil
}
```